Add file-sync support to subsidies consumer

aatauil opened the pull request:
## ID
 DGS-84

 ## Description

When syncing subsidy data from loket, we currently do not get the files yet. This change goes and fetches the files from subsidies if needed. 

 ## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [ ] Other

 ## How to test
 
 Testing exists of 2 parts, initial sync & subsequent delta syncs. 

**initial sync**
- setup clean instance of app-digitaal-loket
- create some subsidie-formulieren in the app (preferably LEKP 1.0 & fietssubsidy) 
- reset inital sync & create new dump file, see the 3 commands in the wiki
```ex
  match "/job-initiator/*path" do
    forward conn, path, "http://delta-producer-background-jobs-initiator-subsidies/"
  end
```
https://github.com/lblod/app-subsidiedatabank/wiki/How-to-setup#initial-sync-process 
- wait for dump to generate
- setup clean app-subsidiedatabank instance, when synced you should now see the same files in both app-digitaal-loket as the subsidiedatabank:
- be sure to check meta-files as well (table data gets generated into meta files) 

**delta sync**
- if initial sync is ok, go to app-digitaal-loket and create a new subsidy
- after a couple of minutes check if files (bijlages) are also visible in subsidiedatabank
- do this for different file-types & subsidies
- be sure to check meta-files as well (table data gets generated into meta files)

aatauil committed:
(credits @wolfez) add subsidies-consumer files

aatauil committed:
mount subsidies-consumer dispatching folder

aatauil committed:
mount files folder

same location as in loket so that subsidies knows where to get the files

aatauil committed:
move file logic to new file-processor file

aatauil committed:
update initial & delta sync file code to call file-processor

aatauil committed:
seperate into delete & insert function

aatauil committed:
termObjects could be empty

wolfderechter commented on config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js:
```diff
@@ -0,0 +1,203 @@
+const DOWNLOAD_OPERATION = "download"
+const DELETE_OPERATION = "delete"
+
+// Cookie nescessary to download file from producer
```
```suggestion
// Cookie necessary to download file from producer
```

wolfderechter commented on config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js:
```diff
@@ -0,0 +1,203 @@
+ */
+function deleteFile(uri) {
+  uri = uri.replace(/[<>]/g, "");
+  const fileName = uri.replace('share://', '');
+  const filePath = `/share/${fileName}`;
```
```suggestion
  const filePath = uri.replace('share://', '/share/');
```
we don't need filename

aatauil replied:
Indeed we dont

wolfderechter committed:
add caching

wolfderechter commented:
I added the caching from subsidy-detailview branch: https://github.com/lblod/app-subsidiedatabank/pull/11/commits/dcc5a2718a408bc3b5a353befaf6c08ae189b488.

LGTM

aatauil committed:
Update config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js

Co-authored-by: wolfderechter <60930264+wolfderechter@users.noreply.github.com>

aatauil committed:
Update config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js

Co-authored-by: wolfderechter <60930264+wolfderechter@users.noreply.github.com>

aatauil committed:
Revert "add caching"

This reverts commit dcc5a2718a408bc3b5a353befaf6c08ae189b488.

aatauil commented:
> I added the caching from subsidy-detailview branch: [dcc5a27](https://github.com/lblod/app-subsidiedatabank/commit/dcc5a2718a408bc3b5a353befaf6c08ae189b488).
> 
> LGTM

Adding caching is ok, just does not belong in this PR. Best to make a seperate PR for that, reverted the change
